play-json/shared/src/main/scala/Reads.scala

```diff
@@ -129,6 +129,22 @@ object Reads extends ConstraintReads with PathReads with DefaultReads with Gener
   }
 
   implicit val JsArrayReducer = Reducer[JsValue, JsArray](js => JsArray(Array(js)))
+
+  /**
+   * A reads that fails if any of the objects properties are not present in the passed in known properties list.
+   */
+  def failOnUnknownProperties(knownProperties: immutable.Seq[String]): Reads[JsObject] = Reads { jsValue =>
```

Comment:
- Directly `JsObjectReads.flatMap`?
- Does the order of properties is/need to be checked? If not maybe passing them as `Set` would make sense

Reply:
`failOnUnknownProperty` for naming (if fails on the first one) ?